net-mgmt/pfSense-pkg-snmptt/files/usr/local/pkg/snmptt.inc

```diff
@@ -0,0 +1,179 @@
+<?php
+/*
+ * snmptt.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2017 Rubicon Communications, LLC (Netgate)
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+require_once("util.inc");
+require_once("functions.inc");
+require_once("pkg-utils.inc");
+require_once("globals.inc");
+
+function php_install_snmptt() {
+	sync_package_snmptt();
+}
+
+function php_deinstall_snmptt() {
+	global $config, $g;
+
+	define('SNMPTT_BASE', '/usr/local');
+
+	unlink_if_exists(SNMPTT_BASE . "/etc/rc.d/snmptt.sh");
+}
+
+function validate_input_snmptt($post, &$input_errors) {
+
+	if (isset($post['snmpttenabled'])) {
+
+		if (!empty($post['snmptt_system_name']) && !is_hostname($post['snmptt_system_name'])) {
+			$input_errors[] = "'SNMPTT System Name' is not a valid hostname.";
+		}
+
+		if (!preg_match("/^\".+\"$/", $post['wildcard_expansion_separator'])) {
+			$input_errors[] = "'Wildcard Expansion Separator' must be within quotes and can contain 1 or more characters.";
+		}
+
+		if (!is_numericint($post['threads_max'])) {
+			$input_errors[] = "'Threads Max' value is not numeric.";
+		}
+
+		if (!is_numericint($post['sleep'])) {
+			$input_errors[] = "'Sleep' value is not numeric.";
+		}
+
+		if (!is_numericint($post['statistics_interval'])) {
+			$input_errors[] = "'Statistics Interval' value is not numeric.";
+		}
+
+	}
+}
+
+function sync_package_snmptt() {
+	global $config, $g;
+
+	conf_mount_rw();
+
+	if (is_service_running('snmptt') && !file_exists("{$g['tmp_path']}/.rc.start_packages.running")) {
+		log_error("Stopping service snmptt");
+		stop_service('snmptt');
+	}
+
+	define('SNMPTT_BASE', '/usr/local');
+
+	if (is_array($config['installedpackages']['snmptt'])) {
+		$snmptt_config = $config['installedpackages']['snmptt']['config'][0];
+		if ($snmptt_config['snmpttenabled'] == "on") {
+			$snmptt_system_name = $snmptt_config['snmptt_system_name'];
+			$mode = $snmptt_config['mode'];
+			$multiple_event = $snmptt_config['multiple_event'];
+			$dns_enable = $snmptt_config['dns_enable'];
+			$strip_domain = $snmptt_config['strip_domain'];
+
+			$strip_domain_list = base64_decode($snmptt_config['strip_domain_list']);
+			$strip_domain_list = preg_replace("~\r\n~", "\n", $strip_domain_list);
+
+			$resolve_value_ip_addresses = $snmptt_config['resolve_value_ip_addresses'];
```

Comment:
Why are you copying all of these to single use variables? Just use the original references directly.